lib/freno/client.rb

```diff
@@ -150,6 +151,8 @@ def decorate(request, with:)
         requests = Array(request)
       end
 
+      validate_and_register_decorators(with)
```

Comment:
as we are only validating, in here, and not registering the decorators per-se, would you mind renaming this to something like 

```suggestion
      validate!(with)
```